core/src/test/java/io/undertow/server/handlers/encoding/RequestContentEncodingTestCase.java

```diff
@@ -103,6 +111,14 @@ public void testDeflateEncoding() throws IOException {
         runTest(sb.toString(), "deflate");
         runTest("Hello World", "deflate");
 
+        // the default buffer size used for tests is 8kb
+        // ensure we send enough data to require multiple buffer allocations when deflating
+        Random random = new Random();
+        byte[] randomBytes = new byte[65536];
+        random.nextBytes(randomBytes);
+        Base64.Encoder encoder = Base64.getEncoder();
+        String randomString = encoder.encodeToString(randomBytes);
+        runTest(randomString, "deflate");
```

Comment:
these tests fail on 2.3.19.Final and pass on 2.3.18.Final (and with this fix).

Reply:
great! Thanks for adding a test @bjlaub !